### PR TITLE
Fix: Remove invalid memoization for MetaData

### PR DESCRIPTION
**Pull request by mbargull**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

This adds the crucial bit of https://github.com/conda/conda-build/pull/4665 as a patch until a new version with the changes from that PR is released.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -7,10 +7,12 @@ package:
 
 source:
   url: https://github.com/conda/{{ name }}/archive/refs/tags/{{ version }}.tar.gz
-  sha256: e33cf2dd0fecb887d50d7c0de13902c9e70976016b6eae336e8896892ae57d67
+  sha256: ef14dbb16179b47bef4fa6860c6467a980afa4afc7ff00521ed5c8dd358dc01a
```
Comment (mbargull) [inline recipe/meta.yaml:10]: I don't know why the checksum changed.
The tag doesn't seem to have moved and the date/time of the release was before the last rebuild.
I thought, but maybe misremembered, that GitHub's tarball generation got more stable at some point in the last years..

Comment (chrisburr) [reply recipe/meta.yaml:10]: I don't think GitHub (and git itself) has ever offered any guarantees about the reproducibility of these auto generated tarballs.

Comment (mbargull) [reply recipe/meta.yaml:10]: Yep, therefore it's okay for them to change (no guarantees). I just felt (likely wrongly so) it doesn't happen as frequently anymore.
I was surprised it changed from only 4 days ago (maybe some tooling/workflow updates/changes at GitHub?).
Then again... why don't we have generated release tarballs for `conda-build` any more? I'm opening an issue for that upstream.

Comment (mbargull) [reply recipe/meta.yaml:10]: > I'm opening an issue for that upstream.

https://github.com/conda/conda-build/issues/4667

**Comment by mbargull** [comment]
I'm going ahead with the merge here and will follow up once upstream is updated too.

**Pull request merged**